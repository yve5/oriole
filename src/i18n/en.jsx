const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  'About us': 'About us',
  'An error has been detected. Please try again.':
    'An error has been detected. Please try again.',
  Buttons: 'Buttons',
  Changelog: 'Changelog',
  Collapse: 'Collapse',
  Expand: 'Expand',
  Internationalization: 'Internationalization',
  Paging: 'Paging',
  'The request has been sent successfully.':
    'The request has been sent successfully.',
};

export default I18N_DICTIONARY;
