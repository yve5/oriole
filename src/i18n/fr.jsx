const I18N_DICTIONARY = {
  LANG: 'fr',
  LANGUAGE: 'Français',
  DATE_FORMAT: 'dd/MM/yyyy',
  LOCALE: 'fr',
  'About us': 'À propos de nous',
  'An error has been detected. Please try again.':
    'Une erreur a été détectée. Merci de recommencer.',
  Buttons: 'Boutons',
  Changelog: 'Journal des modifications',
  'Changes have been saved.': 'Les changements ont été sauvegardés.',
  'Choose a file': 'Choisir un fichier',
  Collapse: 'Réduire',
  Columns: 'Colonnes',
  'Creation date': 'Date de création',
  'Descending order': 'Ordre décroissant',
  Disabled: 'Désactivé',
  Enabled: 'Activé',
  Expand: 'Développer',
  First: 'Premier',
  'Hello World': 'Bonjour le monde',
  'Hide columns': 'Cacher les colonnes',
  Internationalization: 'Internationalisation',
  Paging: 'Pagination',
  'The request has been sent successfully.':
    'La demande a été envoyée avec succès.',
};

export default I18N_DICTIONARY;
