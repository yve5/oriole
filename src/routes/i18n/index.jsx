import I18nPage from './components/I18n';

export default I18nPage;
