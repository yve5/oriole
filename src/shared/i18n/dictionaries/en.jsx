const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  '$number items per page': '$number items per page',
  '$total records': '$total records',
  Add: 'Add',
  'Ascending order': 'Ascending order',
  Back: 'Back',
  'Changes have been saved.': 'Changes have been saved.',
  'Choose a file': 'Choose a file',
  Columns: 'Columns',
  'Creation date': 'Creation date',
  'Descending order': 'Descending order',
  Disabled: 'Disabled',
  Enabled: 'Enabled',
  First: 'First',
  'Hello World': 'Hello World',
  'Hide columns': 'Hide columns',
  Last: 'Last',
  'Loading...': 'Loading...',
  Members: 'Members',
  Next: 'Next',
  No: 'Non',
  'No record': 'No record',
  'One record': 'One record',
  Previous: 'Previous',
  Reset: 'Reset',
  'This field is invalid.': 'This field is invalid.',
  Title: 'Title',
  'to fill': 'to fill',
  'Update of $update, created at $start by $guest':
    'Update of $update, created at $start by $guest',
  Yes: 'Oui',
};

export default I18N_DICTIONARY;
