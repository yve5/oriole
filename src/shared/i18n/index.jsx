import LangSwitch from './components/LangSwitch';
import I18n from './components/I18n';
import i18nReducer from './reducer';

export { I18n, LangSwitch, i18nReducer };
