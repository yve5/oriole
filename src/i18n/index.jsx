import fr from './fr';
import en from './en';
import { tsl } from '../shared/i18n/utils/Translate';

const i18n = {
  tsl: (term, args) => tsl(fr, term, args),
  dictionaries: [fr, en],
  lang: fr.LANG,
};

export default i18n;
