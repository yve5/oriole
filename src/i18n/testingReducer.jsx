import i18n from '../shared/i18n/reducer';
import fr from './fr';
import en from './en';

export default i18n([fr, en])();
