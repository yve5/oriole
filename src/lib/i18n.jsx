import { I18n, LangSwitch, i18nReducer } from '../shared/i18n';

export { I18n, LangSwitch, i18nReducer };
