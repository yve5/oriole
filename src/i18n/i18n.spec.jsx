import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import i18n from '.';
import en from './en';
import fr from './fr';

import { I18n } from '../shared/i18n';
import {
  trimTerms,
  checkMissingTerms,
  checkLocalTermOrder,
  convertLocalCollection,
} from '../shared/i18n/utils';

describe('Local i18n', () => {
  it('should translate', () => {
    expect(i18n.tsl('Hello World')).toEqual('Bonjour le monde');
  });

  it('should detect missing terms', () => {
    expect(checkMissingTerms(en, fr)).toEqual([]);
  });

  it('should check english term order', () => {
    expect(checkLocalTermOrder(en)).toEqual(convertLocalCollection(en));
  });

  it('should check french term order', () => {
    expect(checkLocalTermOrder(fr)).toEqual(convertLocalCollection(fr));
  });

  it('should trim local dictionaries', () => {
    expect(trimTerms(en)).toEqual([]);
  });

  it('should match expected snapshot', () => {
    const mockStore = configureMockStore([thunk]);

    const { asFragment } = render(
      <Provider store={mockStore({ i18n })}>
        {['Hello World'].map((term, index) => (
          <p key={index}>
            <I18n>{term}</I18n>
          </p>
        ))}
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
