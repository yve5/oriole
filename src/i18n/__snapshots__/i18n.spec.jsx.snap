// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Local i18n > should match expected snapshot 1`] = `
<DocumentFragment>
  <p>
    Les changements ont été sauvegardés.
  </p>
  <p>
    La demande a été envoyée avec succès.
  </p>
  <p>
    Une erreur a été détectée. Merci de recommencer.
  </p>
</DocumentFragment>
`;
